test/extended/operators/olm.go

```diff
@@ -71,3 +71,57 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 		})
 	}
 })
+
+// This context will cover test case: OCP-23440, author: jiazha@redhat.com
+var _ = g.Describe("[Feature:Platform] the end user use OLM", func() {
+	defer g.GinkgoRecover()
+
+	var oc = exutil.NewCLI("olm-23440", exutil.KubeConfigPath())
+
+	var (
+		buildPruningBaseDir = exutil.FixturePath("testdata", "olm")
+		operatorGroup       = filepath.Join(buildPruningBaseDir, "operatorgroup.yaml")
+		catalogSourceConfig = filepath.Join(buildPruningBaseDir, "csc.yaml")
+		etcdSub             = filepath.Join(buildPruningBaseDir, "etcd-subscription.yaml")
+		etcdCluster         = filepath.Join(buildPruningBaseDir, "etcd-cluster.yaml")
+	)
+
+	files := []string{operatorGroup, catalogSourceConfig, etcdSub}
+	g.It(" subscribe the single etcd operator", func() {
+		g.By("Cluster-admin user subscribe the operator resource")
+		for _, v := range files {
+			configFile, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", v, "-p", "NAME=test-operator", fmt.Sprintf("NAMESPACE=%s", oc.Namespace()), "MARKETPLACE=openshift-marketplace").OutputToFile("config.json")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = oc.AsAdmin().WithoutNamespace().Run("create").Args("-f", configFile).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+		}
+		time.Sleep(90 * time.Second)
+		output, err := oc.AsAdmin().Run("get").Args("-n", oc.Namespace(), "csv", "etcdoperator.v0.9.4", "-o=jsonpath={.status.phase}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(output).To(o.Equal("Succeeded"))
+
+		g.By("Switch to common user to create the resources provided by the operator")
+		etcdClusterName := "example-etcd-cluster"
+		configFile, err := oc.Run("process").Args("-f", etcdCluster, "-p", fmt.Sprintf("NAME=%s", etcdClusterName), fmt.Sprintf("NAMESPACE=%s", oc.Namespace())).OutputToFile("config.json")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.Run("create").Args("-f", configFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		time.Sleep(90 * time.Second)
```

Comment:
Same - can we poll or watch the etcdcluster's status?

Reply:
@ecordell Thanks! Replace it with `wait.Poll`.